NewDawn/NewDawn/MainPageViewController.swift

```diff
@@ -38,4 +40,75 @@ class MainPageViewController: UIViewController {
             self.performSegue(withIdentifier: "mainPageSelf", sender: nil)
         }
     }
+    
+    func getUserData(handleComplete:(()->())){
+        let request = createGetProfileRequest()
+        processSessionTasks(request: request!, callback: readReponse)
+        handleComplete()
+    }
+    
+    func readReponse(response: NSDictionary) -> Void{
+        let profile_responses = response["objects"] as? [[String: Any]]
+        for profile in profile_responses!{
+            let dummy_user = getProfileInfo(profile: profile)
+            USER_DUMMY_DATA.append(dummy_user as NSDictionary)
+        }
+    }
+
+    func encodeParams(raw_params: Dictionary<String, String>) -> String{
+        var params:String = ""
+        var n = 0
+        for (key, value) in raw_params{
+            if n == 0{
+                params = "?" + key + "=" + value
+            }else{
+                params = params + "&" + key + "=" + value
+            }
+            n += 1
+        }
+        return params
+    }
+    
+    func createGetProfileRequest() -> URLRequest? {
+        // TODO: get username and api_key from keychain/local storage
+        let psudo_params = [
+            "username":"duckmoll",
+            "api_key":"f1c085c2ed2196cb029827c80f879a2ef3ce2189"
+        ]
+        let params = encodeParams(raw_params: psudo_params)
+        let url = getURL(path: "profile/" + params)
+        var request = URLRequest(url:url)
+        request.httpMethod = "GET"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("ApiKey", forHTTPHeaderField: "Authorization")
+        return request
+    }
+    
+    func getProfileInfo(profile: [String: Any]) -> [String: Any]{
+        let user_data = profile["user"]! as? [String: Any]
+        var info = [
+            FIRSTNAME: user_data![FIRSTNAME]! as! String,
+            LASTNAME: user_data![LASTNAME]! as! String,
+            DEGREE: profile[DEGREE]! as! String,
+            SCHOOL: profile[SCHOOL]! as! String,
+            "images": [
+                [
+                    "media": "media/images/testcat.JPG",
+                    "caption": "First image"
+                ],
+            ],
+            "question_answers":[String]()
+            ] as [String : Any]
+        for answer_question in (profile["answer_question"] as? [[String: Any]])!{
+            let answer_question_dict = [
+                "id": answer_question["order"]!,
+                QUESTION: answer_question[QUESTION]!,
```

Comment:
I will change QUESTION and ANSWER.
QUESTION_ANSWERS: front end defines it "question_answers", but back end defines it "answer_question"
ID: ID is a little too big for picture ids as global name